internal/service/networkfirewall/rule_group.go

```diff
@@ -58,6 +58,46 @@ func ResourceRuleGroup() *schema.Resource {
 				Computed: true,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
+						"reference_sets": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"ip_set_references": {
+										Type:     schema.TypeSet,
+										Optional: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"key": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: validation.All(
+														validation.StringLenBetween(1, 32),
+														validation.StringMatch(regexp.MustCompile(`^[A-Za-z]`), "must begin with alphabetic character"),
+														validation.StringMatch(regexp.MustCompile(`^[A-Za-z0-9_]+$`), "must contain only alphanumeric and underscore characters"),
+													),
+												},
+												"ip_set_reference": {
+													Type:     schema.TypeList,
+													Required: true,
+													MaxItems: 5,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"reference_arn": {
+																Type:     schema.TypeString,
+																Required: true,
+																//Elem:     &schema.Schema{Type: schema.TypeString},
```

Comment:
Please remove this comment, and add validation
```suggestion
				ValidateFunc: verify.ValidARN,
```

Reply:
Changed as requested.